nomad/state/state_store_test.go

```diff
@@ -4277,27 +4277,6 @@ func TestStateStore_CSIPlugin_Lifecycle(t *testing.T) {
 		require.True(t, plug.ControllerRequired)
 		require.False(t, plug.IsEmpty())
 
-		updateAllocsFn(allocIDs, SERVER,
-			func(alloc *structs.Allocation) {
-				alloc.DesiredStatus = structs.AllocDesiredStatusStop
-			})
-
-		updateAllocsFn(allocIDs, CLIENT,
-			func(alloc *structs.Allocation) {
-				alloc.ClientStatus = structs.AllocClientStatusComplete
-			})
-
-		plug = checkPlugin(pluginCounts{
-			controllerFingerprints: 1,
-			nodeFingerprints:       2,
-			controllersHealthy:     1,
-			nodesHealthy:           2,
-			controllersExpected:    0,
-			nodesExpected:          0,
-		})
-		require.True(t, plug.ControllerRequired)
-		require.False(t, plug.IsEmpty())
```

Comment:
I think I'm a bit confused here. We can't call `updateAllocsFn` here on any alloc, because this will cause a panic, all allocs for the job have been removed by `store.DeleteJob`.

Reply:
I said `Node.UpdateStatus` above and I should have said `Node.UpdateAlloc`. My concern was what would happen if we got a `Node.UpdateAlloc` RPC concurrently with the purge, such that the validation checks in the RPC handler happened before the deletion, but then the RPC's Raft entry was operating on the now-deleted job. But I've had a quick check through the FSM + state code for that code path and I'm reasonably confident that's safe.

Changing the `updateAllocsFn` would probably have ripple effects that would make the rests of the test awkward. I might argue it'd be worth trying to reproduce the scenario I described above, but that would be adding a completely separate concern to this test than the plugin counts. So this is probably fine as you've got it.